### ISSUE TITLE
NVDA does not read in the Geekbench 6 utility

### ISSUE THREAD

**Issue by enessaribas** [nvaccess/nvda#16251]
### Steps to reproduce:
1. download and install geekbench
2. open the application and try to press alt to access the ribbon
### Actual behavior:

 NVDA will not read the program and will not read options in ribbon. Jaws is able to work with the application. The following errors generate in log

ERROR - core.CorePump.Notify (04:18:35.343) - MainThread (5648):
errors in this core pump cycle
Traceback (most recent call last):
  File "core.pyc", line 837, in Notify
  File "IAccessibleHandler\__init__.pyc", line 1063, in pumpAll
  File "IAccessibleHandler\__init__.pyc", line 725, in processFocusWinEvent
  File "IAccessibleHandler\__init__.pyc", line 560, in winEventToNVDAEvent
  File "UIAHandler\__init__.pyc", line 1242, in isUIAWindow
  File "UIAHandler\__init__.pyc", line 1147, in _isUIAWindowHelper
  File "baseObject.pyc", line 41, in __get__
  File "appModuleHandler.pyc", line 526, in _get_productName
  File "appModuleHandler.pyc", line 521, in _setProductInfo
  File "appModuleHandler.pyc", line 480, in _getExecutableFileInfo
  File "fileUtils.pyc", line 67, in funcWrapper
  File "fileUtils.pyc", line 82, in getFileVersionInfo
RuntimeError: No version information
ERROR - core.CorePump.Notify (04:18:35.354) - MainThread (5648):
errors in this core pump cycle
Traceback (most recent call last):
  File "core.pyc", line 837, in Notify
  File "IAccessibleHandler\__init__.pyc", line 1063, in pumpAll
  File "IAccessibleHandler\__init__.pyc", line 725, in processFocusWinEvent
  File "IAccessibleHandler\__init__.pyc", line 560, in winEventToNVDAEvent
  File "UIAHandler\__init__.pyc", line 1242, in isUIAWindow
  File "UIAHandler\__init__.pyc", line 1147, in _isUIAWindowHelper
  File "baseObject.pyc", line 41, in __get__
  File "appModuleHandler.pyc", line 526, in _get_productName
  File "appModuleHandler.pyc", line 521, in _setProductInfo
  File "appModuleHandler.pyc", line 480, in _getExecutableFileInfo
  File "fileUtils.pyc", line 67, in funcWrapper
  File "fileUtils.pyc", line 82, in getFileVersionInfo
RuntimeError: No version information
ERROR - comtypes._comobject.call_without_this (04:18:35.364) - Dummy-5 (2776):
Exception in IUIAutomationFocusChangedEventHandler.HandleFocusChangedEvent implementation:
Traceback (most recent call last):
  File "comtypes\_comobject.pyc", line 154, in call_without_this
  File "UIAHandler\__init__.pyc", line 843, in IUIAutomationFocusChangedEventHandler_HandleFocusChangedEvent
  File "UIAHandler\__init__.pyc", line 1399, in isNativeUIAElement
  File "UIAHandler\__init__.pyc", line 1242, in isUIAWindow
  File "UIAHandler\__init__.pyc", line 1147, in _isUIAWindowHelper
  File "baseObject.pyc", line 41, in __get__
  File "appModuleHandler.pyc", line 526, in _get_productName
  File "appModuleHandler.pyc", line 521, in _setProductInfo
  File "appModuleHandler.pyc", line 480, in _getExecutableFileInfo
  File "fileUtils.pyc", line 67, in funcWrapper
  File "fileUtils.pyc", line 82, in getFileVersionInfo
RuntimeError: No version information
ERROR - core.CorePump.Notify (04:18:35.372) - MainThread (5648):
errors in this core pump cycle
Traceback (most recent call last):
  File "core.pyc", line 837, in Notify
  File "IAccessibleHandler\__init__.pyc", line 1059, in pumpAll
  File "IAccessibleHandler\__init__.pyc", line 650, in processGenericWinEvent
  File "IAccessibleHandler\__init__.pyc", line 560, in winEventToNVDAEvent
  File "UIAHandler\__init__.pyc", line 1242, in isUIAWindow
  File "UIAHandler\__init__.pyc", line 1147, in _isUIAWindowHelper
  File "baseObject.pyc", line 41, in __get__
  File "appModuleHandler.pyc", line 526, in _get_productName
  File "appModuleHandler.pyc", line 521, in _setProductInfo
  File "appModuleHandler.pyc", line 480, in _getExecutableFileInfo
  File "fileUtils.pyc", line 67, in funcWrapper
  File "fileUtils.pyc", line 82, in getFileVersionInfo
RuntimeError: No version information

### Expected behavior:

NVDA reads the program and reads options in ribbon

#### NVDA installed/portable/running from source:
installed
#### NVDA version:
NVDA version alpha-31229,74a19985
#### Windows version:
windows 11 23h2 (OS Build 22631.3235)
#### Name and version of other software in use when reproducing the issue:
Geekbench 6
#### Does the issue still occur after restarting your computer?
yes
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
no
#### If NVDA add-ons are disabled, is your problem still occurring?
yes
#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
yes

**Comment by Adriani90** [comment]
Cc: @LeonarddeR

**Comment by Emil-18** [comment]
This seams to happen because appModule._getExecutableFileInfo raises a "RuntimeError: No version information"exception in Geekbenche. I have seen this error in other applications as well, e.g sasm, but it didn't cause any problems for NVDA there. What about just returning empty strings in _getExecutableFileInfo when it fails?

**Comment by lukaszgo1** [comment]
@Emil-18  Is the version information for sasm's executable visible in its properties in Windows Explorer? If it is then NVDA should probably be able to  retrieve it in some way.

**Comment by Emil-18** [comment]
@lukaszgo1 It isn't visible

**Comment by mzanm** [comment]
Hello. The issue isn't exactly that it doesn't have version information, many apps do, but the issue is that Geekbench uses the window class name "NetUIHWND" which is commonly seen in MS Office apps and MS Office apps  always have version information. Here's the code i'm referring to.

```
		if windowClass == "NetUIHWND" and appModule:
...
			isOfficeApp = appModule.productName.startswith(("Microsoft Office", "Microsoft Outlook"))
			isOffice2013OrOlder = int(appModule.productVersion.split(".")[0]) < 16
```

so it assumes that productName and productVersion will not raise an exception, will return a string, and productVersion will be a string that's like an integer.
I'm unsure of what's the best way to fix this, should fileUtils.getVersionInfo not raise an exception, or just wrap the code above in a try except block and leave everything else as is? I can confirm that wrapping the office check with a try block fixes the issue and Geekbench can be used normally without issue.

**Comment by Emil-18** [comment]
@mzanm I think the best solution may be to put the bit in appModule._getExecutableFileInfo that uses fileUtils.getVersionInfo in a try block, and return ("", "0") if it fails. It could probably also work to not raise an exception in getFileInfo. I don't think properties like productName and productVersion is supposed to raise an exception like they do now

**Comment by mzanm** [comment]
I think that might be API-breaking, because add-ons expect product name and product version to raise an exception, and changing the behavior to returning a string that's "0" might be undesirable, especially when it's possible for "0" to technically be a valid version number. I think it might break the addon "say product name and version" into returning 0 as a version number for every app with an invalid version info.

**Comment by Emil-18** [comment]
@mzanm Ok. I was thinking what if this happens in other situations as well, but your first suggestion could be implemented for now, and the API braking stuff could be implemented in NVDA 2025.1 instead.

**Comment by XLTechie** [comment]
> I think that might be API-breaking, because add-ons expect product name and product version to raise an exception, and changing the behavior to returning a
> string that's "0" might be undesirable, especially when it's possible for "0" to technically be a valid version number. I think it might break the addon
> "say product name and version" into returning 0 as a version number for every app with an invalid version info.

As the maintainer of that add-on, I agree that it would. It could easily be 
fixed, of course, but I would not like to see false zeros returned for versions 
that were not actually zero.
